lua/autorun/client/cl_chip_lister.lua

```diff
@@ -0,0 +1,308 @@
+    y = y + FONT_SIZE * 2
+
+    surface.SetTextPos( x, y )
+    surface.DrawText( STR_GLOBAL )
+    surface.SetTextColor( COLOR_TEXT_FADED )
+    surface.DrawText( globalUsageStrLead )
+    surface.SetTextColor( COLOR_TEXT )
+    surface.DrawText( globalUsageStr )
+    surface.SetTextColor( COLOR_MICROS )
+    surface.DrawText( STR_MICROSECONDS )
+    y = y + FONT_SIZE
+
+    for i = 1, plyCount do
+        local data = rawget( perPlyData, i )
+        local dataCount = rawget( data, "Count" )
+        local ownerUID = rawget( data, "OwnerUID" )
+        local ownerUsage = rawget( data, "OwnerUsage" )
+
+        local owner = ownerUID == ID_WORLD and ID_WORLD or getPlayerByUID( ownerUID )
+        local ownerColor = getTeamColor( owner)
+        local ownerColorFaded = fadeColor( ownerColor )
+        local ownerUsageStrLead, ownerUsageStr = formatCPUs( ownerUsage )
+
+        surface.SetTextPos( x, y )
+        surface.SetTextColor( ownerColor )
+        surface.DrawText( rawget( data, "OwnerName" ) )
+
+        surface.SetTextPos( xEnd + INFO_OFFSET_OWNER, y )
+        surface.DrawText( STR_TOTAL )
+        surface.SetTextColor( ownerColorFaded )
+        surface.DrawText( ownerUsageStrLead )
+        surface.SetTextColor( ownerColor )
+        surface.DrawText( ownerUsageStr )
+        surface.SetTextColor( COLOR_MICROS )
+        surface.DrawText( STR_MICROSECONDS )
+        y = y + FONT_SIZE
+        elemCount = elemCount + 1
+
+        if elemCount == MAX_ELEMENTS then
+            x = SCREEN_SIZE_HALF
+            xEnd = xEnd + SCREEN_SIZE_HALF
+            y = FONT_SIZE * 4
+        end
+
+        for i2 = 1, dataCount / 3 do
+            local baseInd = i2 * 3 - 2
+            local chipShorthand = rawget( data, baseInd + 1 )
+            local chipUsageStrLead, chipUsageStr = formatCPUs( rawget( data, baseInd + 2 ) )
+
+            surface.SetTextPos( x, y )
+            surface.SetTextColor( ownerColor )
+            surface.DrawText( rawget( data, baseInd ) ) -- chipName
+
+            surface.SetTextPos( xEnd + INFO_OFFSET_CHIP, y )
+            surface.SetTextColor( rawget( CHIP_COLORS, chipShorthand ) )
+            surface.DrawText( chipShorthand .. " " )
+            surface.SetTextColor( COLOR_TEXT_FADED )
+            surface.DrawText( chipUsageStrLead )
+            surface.SetTextColor( COLOR_TEXT )
+            surface.DrawText( chipUsageStr )
+            surface.SetTextColor( COLOR_MICROS )
+            surface.DrawText( STR_MICROSECONDS )
+            y = y + FONT_SIZE
+            elemCount = elemCount + 1
+
+            if elemCount == MAX_ELEMENTS then
+                x = SCREEN_SIZE_HALF
+                xEnd = xEnd + SCREEN_SIZE_HALF
+                y = FONT_SIZE * 4
+            end
+        end
+    end
+
+    surface.SetDrawColor( COLOR_DIVIDER )
+    surface.DrawLine( SCREEN_SIZE_HALF, FONT_SIZE * 3, SCREEN_SIZE_HALF, SCREEN_SIZE )
+
+    cam.End2D()
+    render.PopRenderTarget()
+end )
```

Comment:
I suggest splitting this up into a function that does the rendering, and then further splitting it up to things like `drawChipRow` and things like that